src/cli/commands/install.js

```diff
@@ -295,9 +267,38 @@ export class Install {
         }
       };
 
-      pushDeps('dependencies', {hint: null, optional: false}, true);
-      pushDeps('devDependencies', {hint: 'dev', optional: false}, !this.config.production);
-      pushDeps('optionalDependencies', {hint: 'optional', optional: true}, !this.flags.ignoreOptional);
+      pushDeps('dependencies', projectManifestJson, {hint: null, optional: false}, true);
+      pushDeps('devDependencies', projectManifestJson, {hint: 'dev', optional: false}, !this.config.production);
+      pushDeps(
+        'optionalDependencies',
+        projectManifestJson,
+        {hint: 'optional', optional: true},
+        !this.flags.ignoreOptional,
+      );
+
+      if (this.config.workspacesEnabled) {
```

Comment:
Turns out this condition is still needed for commands that need `fetchRequestFromCwd` but have workspaces disabled.
It would try to add workspaces resolution anyway.